docs/proposals/decentralized-identity-management.md

```diff
@@ -0,0 +1,177 @@
+
+### Why not Kubernetes secrets?
+
+* Kubernetes secrets work in a centralized model of secret distribution. To support offline scenarios, usage of long-lived certificates are used. Usage of long-lived certificates expose keys and certificates for longer duration which will enable theft and therefore are a higher value to an attacker since a stolen key enables the attacker to impersonate the service.
+* Known risks of using Kubernetes secrets (here)[https://kubernetes.io/docs/concepts/configuration/secret/#risks]. The presented solution mitigates risks related to protection of secret by application, secret visibility to other instance of application, impersonation of application instance (though, this requires a stronger method of workload selector used for attestation) and accidental storage of secretes on shared volumes by storing in-memory. In-memory storage have issues but pose a harder problem. A better notification framework will be proposed in the forthcoming versions.
+
+## Architecture goal.
+
+* High-level architecture that is achieved in the presented design.\
+
+<img src="../images/security/SpireEdgev0.1-SpireEdgeIntegrationDetailed.png">
+
+* For reference, a PoC without integration with KubeEdge is present in examples.
+
+https://github.com/kubeedge/examples/blob/master/security-demo/README.md
+
+Demo video can be found in the following link:
+
+https://www.youtube.com/watch?v=Nq9EzTrWTRM&t=24s
+
+### Use Cases
+
+* Perform create/delete operations on node entries.
+  * Users will create and delete node entries with identity framework by adding nodes and application instances either using kubernetes master or spire cli along with manual deployment of application.
+  * Identity framework will perform node attestation based on the entry and attestation plugin configured.
+
+* Perform CRUD operations on workload entries.
+   * Users can create, update and delete application instance.
+   * Identity framework will perform workload attestation and provide application leaf certificates.
+
+* Workload pod liveness/readiness probe report can be used to understand the connectivity.
+
+* Workload implements workload api to retrieve and rotate certificates.
+
+## Design Details
+
+### Node registration and deletion
+
+Node regisrations are created using exiting kubernetes API. Alternatively, node registrations can also be created using spire-server cli.
+
+<img src="../images/security/SpireEdgev0.1-SpireEdgeIntegrationDetailed.png">
+
+On addition of a new edge node, identity controller watches nodes and automatically creates entry for edge node and edge components with attestation information.
+
+For cloud node registration,  
+
++ Master node is created once the kubernetes control plane is initialized. 
+
++ Based on the node type, master or slave, respective edge component are assigned a spiffe id with attestation information as configured.
```

Comment:
Can you please add details on what's included in the attestation info.? Also what's the mechanism for audition?